src/fe-common/core/fe-common-core.c

```diff
@@ -461,26 +462,63 @@ void fe_common_core_finish_init(void)
 
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
+	const WindowType type = dest->window->active ? window_item_get_type(dest->window->active) : WI_TYPE_OTHER;
+	GSList *targets = NULL, *iterator = NULL;
+	gboolean found = FALSE;
+
 	g_return_val_if_fail(array != NULL, FALSE);
 
+	// we ignore all targets
 	if (strarray_find(array, "*") != -1)
 		return TRUE;
-
-	if (strarray_find(array, dest->target) != -1)
+	// exit if not a channel or query
```

Comment:
instead of `strarray_find`,, `strarray_find_prefix` and GSLists, rewrite the code like this:

- loop **once** over array
- in each iteration, check if anything that needs to match, does match

Reply:
example check:
```
if (item != NULL && item->type == channel_type && g_strcmp0(current_element, "#"))
```